models/issues/review.go

```diff
@@ -460,8 +460,10 @@ func SubmitReview(ctx context.Context, doer *user_model.User, issue *Issue, revi
 func GetReviewByIssueIDAndUserID(ctx context.Context, issueID, userID int64) (*Review, error) {
 	review := new(Review)
 
-	has, err := db.GetEngine(ctx).SQL("SELECT * FROM review WHERE id IN (SELECT max(id) as id FROM review WHERE issue_id = ? AND reviewer_id = ? AND original_author_id = 0 AND type in (?, ?, ?))",
-		issueID, userID, ReviewTypeApprove, ReviewTypeReject, ReviewTypeRequest).
+	has, err := db.GetEngine(ctx).Where(
+		builder.In("type", ReviewTypeApprove, ReviewTypeReject, ReviewTypeRequest).
```

Comment:
The two SQL is not equal. Take care of the `max(id)`.

Reply:
It uses `ORDER BY id DESC` to select the highest id.

Reply:
The original SQL means only get every reviewer's latest review. But the new SQL is not even if it has an order by.

Reply:
That's not my main concern.
What I think is more important:
Reviews surely store any sort of timestamp, which we should use instead of the ID.

Reply:
Since it's a Get op, I think I'm wrong. It's the same. Because `Get` will add a `Limit 1` SQL.

Reply:
yes that's it :)